ecosystem/indexer/src/indexer/tailer.rs

```diff
@@ -80,6 +86,48 @@ impl Tailer {
         info!("Migrations complete!");
     }
 
+    /// If chain id doesn't exist, save it. Otherwise make sure that we're indexing the same chain
+    pub async fn check_or_update_chain_id(&self) -> anyhow::Result<bool> {
+        info!("Checking if chain id is correct");
+        let conn = self
+            .connection_pool
+            .get()
+            .expect("DB connection should be available at this stage");
+
+        let query_chain = dsl::ledger_infos
+            .select(dsl::chain_id)
+            .load::<i64>(&conn)
+            .expect("Error loading chain id from db");
+        let maybe_existing_chain_id = query_chain.first();
+
+        let new_chain_id = self
+            .transaction_fetcher
+            .lock()
+            .await
+            .fetch_ledger_info()
+            .await
+            .chain_id as i64;
```

Comment:
Why are we casting this to an i64?

Reply:
To compare against the chain_id we get back from the db.

Reply:
Spoke offline, we'll land this for now but Bowen has pinky promised to fix this later. We have the same problem with the ledger version, we're storing that as i64, which is also incorrect. @bowenyang007 could you link the issue here?

Reply:
@banool we discussed this offline
#2418 https://github.com/aptos-labs/aptos-core/issues/2418